lib/controllers/spending_limit_controller.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:sun_flutter_capstone/models/model.dart';
+
+class SpendingLimitController {
+  Future<Spending_limit?> getCurrentSpendingLimit(
+    DateTime dateNow) async {
+    return await Spending_limit()
+      .select()
+      .start_date
+      .lessThanOrEquals(dateNow)
+      .and
+      .end_date
+      .greaterThanOrEquals(dateNow)
+      .toSingle();
+  }
+
+  Future<int?> upsert(double limitAmount) async {
+    var now = DateTime.now();
+    Spending_limit? spendingLimit = await getCurrentSpendingLimit(now);
+    
+    final result = await Spending_limit(
+      id: spendingLimit?.id,
+      amount: limitAmount,
+      start_date: DateTime.now(),
+      end_date: DateTime.now().add(Duration(days: 30)),
```

Comment:
@AubreyV 
ma'am aubs, I already confirmed it that start and end date are from day 1 and last day of the month. Your current implementation po will cause a problem if user will set spending limit in-between sa month. Please consider po :man-bowing: